jdaviz/tests/test_app.py

```diff
@@ -1,57 +1,141 @@
+              'toolbar': ['g-data-tools', 'g-subset-tools'],
+              'tray': ['g-metadata-viewer',
+                       'g-plot-options',
+                       'g-subset-plugin',
+                       'g-gaussian-smooth',
+                       'g-model-fitting',
+                       'g-unit-conversion',
+                       'g-line-list',
+                       'specviz-line-analysis',
+                       'g-export-plot'],
+              'viewer_area': [{'container': 'col',
+                               'children': [{'container': 'row',
+                                             'viewers': [{'name': 'H',
+                                                          'plot': 'specviz-profile-viewer',
+                                                          'reference': 'h'},
+                                                         {'name': 'K',
+                                                          'plot': 'specviz-profile-viewer',
+                                                          'reference': 'k'}]}]}]}
+
+    class Customviz(Specviz):
+        _default_configuration = config
+        _default_spectrum_viewer_reference_name = 'h'
+
+    viz = Customviz()
+    assert viz.app.get_viewer_reference_names() == ['h', 'k']
+
+    viz.load_spectrum(spectrum1d, data_label='example label')
+    assert not len(viz.app.get_data_from_viewer("h", "non-existent label"))
+
+
+def test_duplicate_data_labels(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    dc = specviz_helper.app.data_collection
+    assert dc[0].label == "test"
+    assert dc[1].label == "test (1)"
+    specviz_helper.load_spectrum(spectrum1d, data_label="test_1")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test")
+    assert dc[2].label == "test_1"
+    assert dc[3].label == "test (2)"
+
+
+def test_duplicate_data_labels_with_brackets(specviz_helper, spectrum1d):
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    specviz_helper.load_spectrum(spectrum1d, data_label="test[test]")
+    dc = specviz_helper.app.data_collection
+    assert len(dc) == 2
+    assert dc[0].label == "test[test]"
+    assert dc[1].label == "test[test] (1)"
+
+
+def test_return_data_label_is_none(specviz_helper):
+    data_label = specviz_helper.app.return_data_label(None)
+    assert data_label == "Unknown"
+
+
+def test_return_data_label_is_image(specviz_helper):
+    data_label = specviz_helper.app.return_data_label("data/path/test.jpg")
+    assert data_label == "test[jpg]"
+
+
+def test_hdulist_with_filename(cubeviz_helper, image_cube_hdu_obj):
+    image_cube_hdu_obj.file_name = "test"
+    data_label = cubeviz_helper.app.return_data_label(image_cube_hdu_obj)
+    assert data_label == "test[HDU object]"
+
+
+@pytest.mark.filterwarnings('ignore:.* is a low contrast image')

```

Comment:
The test never attempts to open the empty file, so this warning will not happen.
```suggestion
```